optimade/models/jsonapi.py

```diff
@@ -55,16 +56,39 @@ class JsonApi(BaseModel):
 class ToplevelLinks(BaseModel):
     """A set of Links objects, possibly including pagination"""
 
+    class Config:
+        extra = "allow"
+
     self: Optional[Union[AnyUrl, Link]] = Field(None, description="A link to itself")
     related: Optional[Union[AnyUrl, Link]] = Field(
         None, description="A related resource link"
     )
 
```

Comment:
Also, `base_url` seems to be an accepted field under top-level `links` (see [here](https://github.com/Materials-Consortia/OPTIMADE/blob/v1.0.0/optimade.rst#json-response-schema-common-fields)). And of course other random fields are allowed as well (but probably needs to be prepended with e.g., `_exmpl_`. However, this doesn't seem to be detailed in the specification?

Reply:
It's a bit unclear IMO, `base_url` is indeed included as an example of a `Links` object you could put in the links response, but I feel it's a free-for-all otherwise... We could certainly update this to validate specifically allowed keys in the future, but I think we would need the impetus to do so from the specification.

Reply:
Yeah, so for now I would probably add `base_url` and otherwise keep it as you now have. Then we can open an issue to discuss this in the spec repo.

Reply:
Hmm, I'm not so sure about adding `base_url` explicitly here, we already have a `base_url` entry in the schema for the links endpoint. In my current reading of the specification, I think you can put any key you want in here, with an appropriate description (with no provider prefix needed), as long as it doesn't clash with the JSON API/pagination fields. I think I'd rather leave this as-is until we can clarify this in the spec.